packages/node/src/integrations/tracing/postgresjs.ts

```diff
@@ -283,27 +505,171 @@ export class PostgresJsInstrumentation extends InstrumentationBase<PostgresJsIns
-        .replace(/--.*?(\r?\n|$)/g, '') // Single line comments
+        // Remove comments first (they may contain newlines and extra spaces)
+        .replace(/--.*$/gm, '') // Single line comments (multiline mode)
         .replace(/\/\*[\s\S]*?\*\//g, '') // Multi-line comments
         .replace(/;\s*$/, '') // Remove trailing semicolons
-        .replace(/\b\d+\b/g, '?') // Replace standalone numbers
-        // Collapse whitespace to a single space
+        // Collapse whitespace to a single space (after removing comments)
         .replace(/\s+/g, ' ')
-        // Collapse IN and in clauses
-        // eg. IN (?, ?, ?, ?) to IN (?)
-        .replace(/\bIN\b\s*\(\s*\?(?:\s*,\s*\?)*\s*\)/g, 'IN (?)')
+        .trim() // Remove extra spaces and trim
+        // Replace standalone numbers and parameterized queries ($1, $2, etc.) BEFORE truncation
+        .replace(/\$\d+/g, '?') // Replace PostgreSQL placeholders ($1, $2, etc.)
+        .replace(/\b\d+\b/g, '?') // Replace standalone numbers
+        // Collapse IN and in clauses (eg. IN (?, ?, ?, ?) to IN (?))
+        .replace(/\bIN\b\s*\(\s*\?(?:\s*,\s*\?)*\s*\)/gi, 'IN (?)')
+        .substring(0, 1024) // Truncate to 1024 characters AFTER sanitization
     );
   }
+
+  /**
+   * Fallback patch for Query.prototype.handle to instrument queries from pre-existing sql instances.
+   * This catches queries from sql instances created BEFORE Sentry was initialized (CJS only).
+   *
+   * Note: Queries from pre-existing instances won't have connection context (database, host, port)
+   * because the sql instance wasn't created through our instrumented wrapper.
+   */
+  private _patchQueryPrototype(moduleExports: {
+    Query: {
+      prototype: {
+        handle: ((...args: unknown[]) => Promise<unknown>) & {
+          __sentry_original__?: (...args: unknown[]) => Promise<unknown>;
+        };
+      };
+    };
+  }): typeof moduleExports {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    const originalHandle = moduleExports.Query.prototype.handle;
+
+    moduleExports.Query.prototype.handle = async function (
+      this: {
+        resolve: unknown;
+        reject: unknown;
+        strings?: string[];
+      },
+      ...args: unknown[]
+    ): Promise<unknown> {
+      // Skip if this query came from an instrumented sql instance (already handled by wrapper)
+      if ((this as Record<symbol, unknown>)[QUERY_FROM_INSTRUMENTED_SQL]) {
+        return originalHandle.apply(this, args);
+      }
+
+      // Skip if we shouldn't create spans
+      if (!self._shouldCreateSpans()) {
+        return originalHandle.apply(this, args);
+      }
+
+      const sanitizedSqlQuery = self._sanitizeSqlQuery(this.strings?.[0]);
+
+      return startSpanManual(
+        {
+          name: sanitizedSqlQuery || 'postgresjs.query',
+          op: 'db',
+        },
+        (span: Span) => {
+          addOriginToSpan(span, 'auto.db.otel.postgres');
```

Comment:
l: maybe I've missed something but this is our instrumentation, right? Or did we fork it from OTel?

```suggestion
          addOriginToSpan(span, 'auto.db.postgresjs');
```